### ISSUE TITLE
Custom type convertToJSValueSQL method is not used when calling find method with fields option

### ISSUE THREAD

**Issue by mandrysek** [mikro-orm/mikro-orm#4622]
**Describe the bug**
Method `convertToJSValueSQL` from custom type is not used when calling `find` method with `fields` option.

**Stack trace**
```
export class PointType extends Type<Point | undefined, string | undefined> {
  convertToDatabaseValue(value: Point | undefined): string | undefined {
    if (!value) {
      return value;
    }

    return `point(${value.latitude} ${value.longitude})`;
  }

  convertToJSValue(value: string | undefined): Point | undefined {
    const m = value?.match(/point\((-?\d+(\.\d+)?) (-?\d+(\.\d+)?)\)/i);

    if (!m) {
      return undefined;
    }

    return new Point(+m[1], +m[3]);
  }

  convertToJSValueSQL(key: string) {
    return `ST_AsText(${key})`;
  }

  convertToDatabaseValueSQL(key: string) {
    return `ST_GeogFromText(${key})`;
  }

  getColumnType(): string {
    return 'geography';
  }
}


const data = await this.repository.find(where, {
  fields: ['location'],
});

// With fields
select "a0"."id", "a0"."location" from "ad" as "a0" order by "a0"."id" asc limit 50

// Without fields
select "a0".*, ST_AsText("a0"."location") as "location" from "ad" as "a0" order by "a0"."id" asc limit 50

```

**To Reproduce**
1. Create custom type
2. Use custom type in the Entity
3. Get data with `find` method and use `fields` option which contains field which has the custom type

**Expected behavior**
Custom type method `convertToJSValueSQL`  should be used when explicitly selecting fields

**Additional context**
N/A

**Versions**

| Dependency | Version |
| - | - |
| node | 20.3.1 |
| typescript | 5.1.6 |
| mikro-orm | 5.7.14 |
| pg | 8.11.1 |

**Comment by B4nan** [comment]
I am able to reproduce this on properties that have different property name and column name, e.g. when you map `extendedPoint` to `extended_point`. When the names match (as with `location`) it seems to be working fine - so if that's the case, I will might need to see a complete repro - but maybe the fix will work for you, so be sure to first try the dev version once its out.

**Comment by robbievandam** [comment]
Thanks, just ran into this myself. Is there any further reason why this happens? And, is there a way to fix it? Currently, I've got two fields which are now underscored in my entity definition, instead of camelCased.

**Comment by B4nan** [comment]
This has been fixed in v5.8, if you have a similar problem that is reproducible with the latest version, please provide a complete repro.
